src/geo.c

```diff
@@ -527,16 +531,69 @@ void georadiusGeneric(client *c, int flags) {
                        (i+1) < remaining &&
                        !(flags & RADIUS_NOSTORE))
             {
-                storekey = c->argv[base_args+i+1];
+                if (!(flags & GEOSEARCH)) storekey = c->argv[base_args+i+1]; // GEOSEARCH flag don't update storekey
                 storedist = 0;
                 i++;
             } else if (!strcasecmp(arg, "storedist") &&
                        (i+1) < remaining &&
                        !(flags & RADIUS_NOSTORE))
             {
-                storekey = c->argv[base_args+i+1];
+                if (!(flags & GEOSEARCH)) storekey = c->argv[base_args+i+1];
```

Comment:
this isn't enough. you'll only get here if you one one extra argument left to parse.
i.e. GEOSEARCHSTORE...STOREDIST won't work if STOREDIST is the last argument.
and also, you have an excess `i++` here.
i think it would be better to have the original "store" and "storedist" arguments be excluded when GEOSEARCH is set, and have a separate "storedist" case for the GEOSEARCH case.